OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -73,6 +73,44 @@ public SlotDropDownOption(string title, string order, Func<bool> selected)
 			dropdown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 167, options, setupItem);
 		}
 
+		public static void ShowPlayerActionDropDown(DropDownButtonWidget dropdown, Session.Slot slot,
+			Session.Client c, OrderManager orderManager, Widget lobby,  Action before, Action after)
+		{
+			Action<bool> okPressed = tempBan => { orderManager.IssueOrder(Order.Command("kick {0} {1}".F(c.Index, tempBan))); after(); };
+			var onClick = new Action(() =>
+			{
+				before();
+
+				Game.LoadWidget(null, "KICK_CLIENT_DIALOG", lobby.Get("TOP_PANELS_ROOT"), new WidgetArgs
+				{
+					{ "clientName", c.Name },
+					{ "okPressed", okPressed },
+					{ "cancelPressed", after }
+				});
+			});
+
+			var options = new List<ShowPlayerActionDropDownOption>
+			{
+				new ShowPlayerActionDropDownOption("Promote to Admin", () => { orderManager.IssueOrder(Order.Command("promote {0}".F(c.Index))); }),
```

Comment:
This allows players who are hosting their own local servers to hand over admin to another player.  We only want to enable this on dedicated servers – this can be fixed similarly to the spectator comment above.

Reply:
What would be the best way to detect this?

Reply:
Good question.  If this was server-side code then it would be easy: we have a `Dedicated` flag on the `Server` class.  This could perhaps be moved to the `Global` class where it would be automatically synced to clients, and then accessed from here as `orderManager.LobbyInfo.GlobalSettings.Dedicated`.